### PR TITLE
Add Program Counter (PC) improvements and bug fixes on store instruct…

### PR DESCRIPTION
**Pull request by cfboy**
…ions method.

### CODE REVIEW

File: assembler.py [hunk 1]
```diff
@@ -70,20 +72,36 @@ def read_source(self):
...
                     # Indicates at what memory location it will begin storing instructions
-                    org_address = f'0x{source[1]}'
-                    self.mem_pointer = int(org_address, 16)
-                    print(f'org: {org_address} mem: {self.mem_pointer}')
+                    org_address = source[1]
+                    if self.is_hex_number(org_address):
+                        self.p_counter = int(org_address, 16)
+                    else:
+                        org_address = f'0x{org_address}'
+                        self.p_counter = int(org_address, 16)
```
Comment (ChristianPerez34) [inline assembler.py]: Found out with @noelvalentin that the if statements are unnecessary
```python
self.p_counter = int(org_address, 16)
```

File: assembler.py [hunk 2]
```diff
@@ -70,20 +72,36 @@ def read_source(self):
     def is_valid_source(self):
         return re.match(r'^.+\.asm$', self.filename)
 
+    def is_hex_number(self, number):
+        return not isinstance(number, int)
+
```
Comment (ChristianPerez34) [inline assembler.py]: No longer needed

File: assembler.py [hunk 3]
```diff
@@ -70,20 +72,36 @@ def read_source(self):
...
-                    print(source)
+                    if source[0].lower() in OPCODE:
+                        # Assign instruction to proper memory location
+                        print('It is an assembly instruction')
+                    else:
+                        print(source)
+
+                    self.p_counter += 2  # Increase Program Counter
+                    print('The current PC is: ' + str(self.p_counter))
+                    # logging.debug('The current PC is: ' + str(self.p_counter))
```
Comment (ChristianPerez34) [inline assembler.py]: Remove this comment

File: assembler.py [hunk 4]
```diff
@@ -1,4 +1,6 @@
-import re
+import re, logging
+
+logging.basicConfig(level=logging.DEBUG)  # for debug purpose.
```
Comment (ChristianPerez34) [inline assembler.py]: No longer necessary

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

**Review by ChristianPerez34 — APPROVED** [review]
Good work!

**Review by noelvalentin — APPROVED** [review]

**Review by alejoreyes96 — APPROVED** [review]

**Pull request merged**